### PR TITLE
Themes: migrate CSS to webpack

### PR DESCRIPTION
**Pull request by jsnajdr**
Step by step, commit by commit, we add `import 'style.scss'` statements, split part of the big monolithic `my-sites/themes/style.scss` into smaller modules, remove legacy or unused CSS rules and fix ESLint warnings (especially `jsx-classname-namespace`.

All steps should be straighforward, with no complications.

### CODE REVIEW

File: client/my-sites/themes/themes-site-selector-modal.jsx [hunk 1]
```diff
@@ -140,11 +145,14 @@ class ThemesSiteSelectorModal extends React.Component {
...
+
 export default connect( state => ( {
 	// We don't need a <QueryTheme /> component to fetch data for the theme since the
 	// ThemesSiteSelectorModal will always be called from a context where those data are available.
 	// FIXME: Since the siteId and themeId are part of the component's internal state, we can't use them
 	// here. Instead, we have to return helper functions.
-	getSiteSlug: siteId => getSiteSlug( state, siteId ),
-	getWpcomTheme: themeId => getTheme( state, 'wpcom', themeId ),
+	getSiteSlug: bindGetSiteSlug( state ),
+	getWpcomTheme: bindGetWpcomTheme( state ),
```
Comment (blowery) [inline client/my-sites/themes/themes-site-selector-modal.jsx]: do you also need to mark this as impure? These functions will no longer change on every run of mapStateToProps, so rendering may get skipped?

Comment (jsnajdr) [reply client/my-sites/themes/themes-site-selector-modal.jsx]: They will change on every run and will return a different function object on each call. We'll have unneeded rerenders on every `state` change, but that was the previous behavior, too. Not easy to fix and out of scope for this PR.

**Review by blowery — APPROVED** [review]
LGTM. Tested both logged in and logged out. Looks the same as production.

**Comment by ockham** [comment]
Thanks a lot, Jarda!

**Pull request merged**